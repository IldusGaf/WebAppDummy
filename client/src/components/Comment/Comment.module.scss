@import '../styleConstants.scss';

.comment {
    display: flex;
    background-color: $elementColor;
    border: 0.1rem solid $elementBorderColor;
    flex-direction: column;
    border-radius: $contentBorderRadius;
    margin: 0.5rem;
    padding: 1rem;
    &__user {
        display: flex;
        align-items: center;
        margin-bottom: 0.4rem;
    }
    &__userImage {
        margin: 0;
    img {
        border-radius: 50%;
        height: 40px;
        width: 40px;
    }
}
    &__userDesc {
        display: flex;
        flex-direction: column;
        margin-left: 0.5rem;
    }
    &__userName {
        display: flex;
    }
    &__userDate {
        color: gray;
    }
    &__desc {
        color: inherit;
    }
    &Dark{
        @include darkContent;
    }
}