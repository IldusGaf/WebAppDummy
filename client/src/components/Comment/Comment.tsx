import React, { useContext } from 'react';
import { format } from 'date-fns';
import { ThemeContext } from '../../contexts/ThemeContext';
import { PostCommentType } from '../../types/dummyAPIResponses';
import classes from './Comment.module.scss';

export const Comment = ({
  message,
  owner,
  publishDate,
}: PostCommentType) => {
  const themeContext = useContext(ThemeContext);
  return (
    <div className={`${classes.comment} ${themeContext.darkTheme ? classes.commentDark : ''}`}>
      <div className={classes.comment__user}>
        {/* <Avatar icon={<UserOutlined />} size="large" /> */}
        <figure className={classes.comment__userImage}><img alt={owner.firstName} src={owner.picture} /></figure>
        <div className={classes.comment__userDesc}>
          <span className={classes.comment__userName}>
            {owner.firstName}
            {' '}
            {owner.lastName}
          </span>
          <span className={classes.comment__userDate}>{publishDate && format(new Date(publishDate), 'dd MMMM yyyy HH:mm')}</span>
        </div>
      </div>
      <span className={classes.comment__desc}>{message}</span>
    </div>
  );
};
