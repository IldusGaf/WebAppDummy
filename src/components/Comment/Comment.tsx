import React, { useContext } from 'react';
import { ThemeContext } from '../../contexts/ThemeContext';
import { PostCommentType } from '../../types/dummyAPIResponses';
import classes from './Comment.module.scss';

export const Comment = ({
  message,
  owner,
  publishDate,
}: PostCommentType) => {
  const themeContext = useContext(ThemeContext);
  return (
    <div className={`${classes.comment} ${themeContext.darkTheme ? classes.commentDark : ''}`}>
      <div className={classes.comment__user}>
        {/* <Avatar icon={<UserOutlined />} size="large" /> */}
        <figure className={classes.comment__userImage}><img alt={owner.firstName} src={owner.picture} /></figure>
        <div className={classes.comment__userDesc}>
          <span className={classes.comment__userName}>
            {owner.firstName}
            {' '}
            {owner.lastName}
          </span>
          <span className={classes.comment__userDate}>{publishDate}</span>
        </div>
      </div>
      <span className={classes.comment__desc}>{message}</span>
    </div>
  );
};
