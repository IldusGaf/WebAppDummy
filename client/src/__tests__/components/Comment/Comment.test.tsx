import React from 'react';
import { shallow } from 'enzyme';
import { Comment } from '../../../components/Comment/Comment';

describe('Comment component test', () => {
  test('should render comment', () => {
    const wrapper = shallow(<Comment message="" owner={{ firstName: '', lastName: '', picture: '' }} publishDate="" />);
    expect(wrapper.find('div.comment')).toHaveLength(1);
  });

  test('should render data', () => {
    const title = 'mr';
    const firstName = 'James';
    const lastName = 'Bond';
    const publishDate = '06 December 2020 00:00';
    const text = 'Blah-blah-blah';
    const wrapper = shallow(<Comment
      owner={{ firstName, lastName, picture: '' }}
      publishDate={publishDate}
      message={text}
    />);
    expect(wrapper.find('span.comment__userName').text()).toBe(`${firstName} ${lastName}`);
    expect(wrapper.find('span.comment__userDate').text()).toBe(publishDate);
    expect(wrapper.find('span.comment__desc').text()).toBe(text);
  });

  test('snapshot test', () => {
    const wrapper = shallow(<Comment message="" owner={{ firstName: '', lastName: '', picture: '' }} publishDate="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
