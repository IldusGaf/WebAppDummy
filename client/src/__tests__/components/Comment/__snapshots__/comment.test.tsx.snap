// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component test snapshot test 1`] = `
<div
  className="comment "
>
  <div
    className="comment__user"
  >
    <figure
      className="comment__userImage"
    >
      <img
        alt=""
        src=""
      />
    </figure>
    <div
      className="comment__userDesc"
    >
      <span
        className="comment__userName"
      >
         
      </span>
      <span
        className="comment__userDate"
      />
    </div>
  </div>
  <span
    className="comment__desc"
  />
</div>
`;
